index.ts

```diff
@@ -0,0 +1,207 @@
+
+  console.log(`Found ${virtualMachines.length} virtual machines:`);
+  virtualMachines.forEach((virtualMachine: Models.VirtualMachine, index: number) => {
+    console.log(`${index}): ${virtualMachine.name}\t\t${virtualMachine.location}\t${virtualMachine.provisioningState}`);
+  });
+
+  return virtualMachines;
+}
+
+async function createVirtualNetwork(
+  networkClient: NetworkManagementClient,
+  resourceGroupName: string,
+  name: string,
+  parameters: NetworkModels.VirtualNetwork = {
+    location: "eastus2",
+    addressSpace: {
+      addressPrefixes: ["10.0.0.0/16"]
+    },
+    subnets: [
+      {
+        name: `sub${name}`,
+        addressPrefix: "10.0.0.0/24"
+      }
+    ]
+  }
+): Promise<NetworkModels.VirtualNetwork> {
+  console.log(`Creating "${name}" virtual network in ${resourceGroupName} resource group in ${networkClient.subscriptionId} subscription`);
+  const virtualNetwork = await networkClient.virtualNetworks.createOrUpdate(resourceGroupName, name, parameters);
+
+  console.log(`Virtual network "${virtualNetwork.name} was created successfully`);
+  return virtualNetwork;
+}
+
+async function createPublicIpAddress(
+  networkClient: NetworkManagementClient,
+  resourceGroupName: string,
+  name: string,
+  parameters: NetworkModels.PublicIPAddress = {
+    location: "eastus2",
+    publicIPAllocationMethod: "Dynamic",
+    dnsSettings: {
+      domainNameLabel: name
+    }
+  }
+): Promise<NetworkModels.PublicIPAddress> {
+  console.log(`Creating "${name}" public IP address in ${resourceGroupName} resource group in ${networkClient.subscriptionId} subscription`);
+  const publicIPAddress = await networkClient.publicIPAddresses.createOrUpdate(resourceGroupName, name, parameters);
+
+  console.log(`Virtual network "${publicIPAddress.name} was created successfully`);
+  return publicIPAddress;
+}
+
+async function createNetworkInterface(
+  networkClient: NetworkManagementClient,
+  resourceGroupName: string,
+  name: string,
+  virtualNetwork: NetworkModels.VirtualNetwork,
+  publicIp: NetworkModels.PublicIPAddress,
+  parameters: NetworkModels.NetworkInterface = {
+    location: "eastus2",
+    ipConfigurations: [
+      {
+        name: name,
+        privateIPAllocationMethod: "Dynamic",
+        subnet: virtualNetwork.subnets![0],
+        publicIPAddress: publicIp
+      }
+    ]
+  }
+): Promise<NetworkModels.NetworkInterface> {
+  console.log(`Creating "${name}" network interface in ${resourceGroupName} resource group in ${networkClient.subscriptionId} subscription`);
+  const networkInterface = await networkClient.networkInterfaces.createOrUpdate(resourceGroupName, name, parameters);
+
+  console.log(`Virtual network "${networkInterface.name} was created successfully`);
+  return networkInterface;
+}
+
+async function createVirtualMachine(
+  computeClient: ComputeManagementClient,
+  resourceGroupName: string,
+  virtualMachineName: string,
+  parameters: Models.VirtualMachine = {
+    location: "eastus2"
```

Comment:
You may want to extract this `location` constant to a variable at the top of the file.